File: solo-leveling-user-service/src/main/java/com/sleepkqq/sololeveling/user/service/service/TaskService.java
package com.sleepkqq.sololeveling.user.service.service;

import com.sleepkqq.sololeveling.user.service.model.Task;
import com.sleepkqq.sololeveling.user.service.repository.TaskRepository;
import java.util.Optional;
import java.util.UUID;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class TaskService {

  private final TaskRepository taskRepository;

  public Task get(UUID id) {
    return find(id).orElseThrow(() -> new IllegalArgumentException("Task not found taskId=" + id));
  }

  public Optional<Task> find(UUID id) {
    return taskRepository.findById(id);
  }

  public Task save(Task task) {
    return taskRepository.save(task);
  }
}

File: solo-leveling-player-service/src/main/kotlin/com/sleepkqq/sololeveling/player/service/service/task/TaskService.kt
package com.sleepkqq.sololeveling.player.service.service.task

import com.sleepkqq.sololeveling.player.model.entity.task.Task
import java.time.LocalDateTime
import java.util.UUID

interface TaskService {
	fun get(id: UUID): Task
	fun find(id: UUID): Task?
	fun updateAll(tasks: Collection<Task>, now: LocalDateTime = LocalDateTime.now())
	fun insert(task: Task): Task
	fun insertAll(tasks: Collection<Task>)
	fun update(task: Task, now: LocalDateTime = LocalDateTime.now()): Task
} 